cmd/config/config_check.go

```diff
@@ -0,0 +1,118 @@
+package config
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"github.com/calyptia/api/types"
+	cfg "github.com/calyptia/cli/config"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const KeyCliHealth = "cli_health"
+
+func NewCmdCheckInstall(c *cfg.Config) *cobra.Command {
+	return &cobra.Command{
+		Use:   "check-install",
+		Short: "Check the current configuration and report any issues",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			CheckInstall(c)
+			return nil
+		},
+	}
+}
+
+func CheckInstall(c *cfg.Config) {
+	ok := true
+	if !checkToken(c) {
+		ok = false
+	}
+	if !checkUrl(c) {
+		ok = false
+	}
+	if !checkSSL(c) {
+		ok = false
+	}
+	var k8sConfig *rest.Config
+	var pass bool
+	if k8sConfig, pass = checkK8sCredentials(); !pass {
+		ok = false
+	}
+	if !checkK8sHealth(k8sConfig) {
+		ok = false
+	}
+	if !ok {
+		fmt.Printf("Calyptia CLI is not ready to use\n")
+		return
+	}
+	err := c.LocalData.Save(KeyCliHealth, "ok")
+	if err != nil {
+		fmt.Printf("Calyptia CLI is ready to use but we couldnt save it health status\n")
+	}
+	fmt.Printf("Calyptia CLI is ready to use\n")
+
+}
+
+func checkToken(c *cfg.Config) bool {
+	if c.ProjectToken == "" {
+		fmt.Printf("You need to set a project token to use Calyptia CLI\n")
+		fmt.Printf("\tTip: you can use the flag --token or the command calyptia config set_token to set your project token\n")
```

Comment:
We have a new line above so could all be a single statement

Reply:
we've one already, let me see if we can find a better option
https://github.com/calyptia/cli/blob/main/.github/workflows/lint.yml#L16

Reply:
Yeah it does not lint spelling though, it may not be doable but I would like to see if there is anything.